dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt

```diff
@@ -275,12 +278,80 @@ internal class RumResourceScope(
         }
     }
 
+    internal fun tryExtractTimings(timingsPayload: Map<String, Any?>?): ResourceTiming? {
```

Comment:
I find it a bit over-engineered. Wouldn't be better just to parse this `Map` as:
```
firstByteStart  ((timingsPayload[FIRST_BYTE_TIMING] as? Map<*,*>)?.get(START_TIME_KEY) as? Number)?.toLong()
```
?
This way we could get rid of all these extra functions and code.

Reply:
I tried. I would need to add it 4 times then and it will trigger `ComplexMethod` rule of `Detekt`, need to suppress then.

Reply:
I see, that explains why having all these methods.

Reply:
LGTM but just as an extra thought if you have time, maybe we could extract all this logic in a separated component ?

Reply:
I agree, probably I will just create Util function for all this (don't want to pollute `Map` namespace with extension function and doubt there is a need to dedicated class).